hibernate-core/src/main/java/org/hibernate/jpa/boot/internal/EntityManagerFactoryBuilderImpl.java

```diff
@@ -1394,4 +1382,51 @@ private void addCacheRegionDefinition(CacheRegionDefinition cacheRegionDefinitio
 			this.cacheRegionDefinitions.add( cacheRegionDefinition );
 		}
 	}
+
+	private <T> T loadSettingInstance(String settingName, Object settingValue, Class<T> clazz) {
+		T instance = null;
+		Class<? extends T> instanceClass = null;
+
+		if ( clazz.isInstance( settingValue ) ) {
```

Comment:
not sure if this should be `if ( clazz.isInstance( settingValue ) ) `or `if ( clazz.isAssignableFrom( settingValue.getClass() ) )`

Reply:
The `isInstance` check allows us to pass an actual `IntegratorProvider` instance that was build programatically. We could support both actual instances and `Class<? extend IntegratorProvider>` that allow us to instantiate the `IntegratorProvider`.

Reply:
@vladmihalcea reverted to` clazz.isAssignableFrom( settingValue.getClass())`